### PR TITLE
Check CORES and MEMORY when starting allpairs_multicore

### PR DESCRIPTION
**Pull request by trshaffer**
I *think* the units are correct. `allpairs_multicore` appears to use bytes for calculating blocksize, while the environment variable seems to be in megabytes.

### CODE REVIEW

**Comment by trshaffer** [comment]
This is for #300

**Comment by dthain** [comment]
`block_size` is not bytes, but simply the number of elements in the array.

**Comment by trshaffer** [comment]
I should have been more clear, inside `block_size_estimate` it falls back to half the system's memory, measured in bytes. I'm not sure if there are any memory-intensive tests for allpairs, but I did test the CPU limit.

**Comment by dthain** [comment]
This is the right approach, but let's factor out the examination of the environment and the querying of hardware properties into a module that returns everything in a resource management structure.

(Since Makelfow and other tools need to do the same thing.)

@btovar can you point @trshaffer in the right direction?

**Comment by btovar** [comment]
@trshaffer:  In dttools/src/rmonitor_poll.[hc] we would add a function such as:

struct rmsummary *rmonitor_measure_host();

The returned rmsummary would have the values of cores, memory, disk, etc. The function rmonitor_measure_process should give you a good idea on how struct rmsummary is created, accessed, etc. Available fields are in dttools/src/rmsummary.h (memory and disk sizes in MB, times in usecs).

**Comment by trshaffer** [comment]
Is this more what you had in mind? Right now `rmonitor_measure_host` only fills in the fields used by allpairs, but that's easy to expand.

**Comment by btovar** [comment]
Nice. It looks good to me.

**Comment by dthain** [comment]
Add the measurement of disk (for a specified directory) into `rmonitor_measure_host` b/c we are going to use that function in makeflow, workqueue, and everywhere else to infer the current operating conditions.

**Comment by trshaffer** [comment]
I *think* that's the correct measurement for this case

File: dttools/src/rmonitor_poll.c [hunk 1]
```diff
@@ -819,10 +820,18 @@ int rmonitor_measure_process_update_to_peak(struct rmsummary *tr, pid_t pid) {
...
+struct rmsummary *rmonitor_measure_host(char *path) {
+	uint64_t free_mem;
+	uint64_t total_mem;
+	int64_t total_disk;
+	int64_t file_count;
 	struct rmsummary *tr = rmsummary_create(-1);
 
+	if(path) {
+		path_disk_size_info_get(path, &total_disk, &file_count);
+		tr->disk = total_disk;
```
Comment (dthain) [inline dttools/src/rmonitor_poll.c]: Do the units match here?

**Comment by trshaffer** [comment]
Ah, my mistake. I was reading the wrong line

**Pull request merged**